bindings/matrix-sdk-ffi/src/platform.rs

```diff
@@ -0,0 +1,53 @@
+#[cfg(target_os = "android")]
+pub use android::*;
+#[cfg(target_os = "ios")]
+pub use ios::*;
+#[cfg(not(any(target_os = "ios", target_os = "android")))]
+pub use other::*;
+
+#[cfg(target_os = "android")]
+mod android {
+    use android_logger::{Config, FilterBuilder};
+    use tracing::log::Level;
+
+    #[uniffi::export]
+    pub fn setup_tracing(filter: Option<String>) {
+        std::env::set_var("RUST_BACKTRACE", "1");
+
+        log_panics::init();
+
+        let mut log_config =
+            Config::default().with_min_level(Level::Trace).with_tag("matrix-rst-sdk");
+        if let Some(filter) = filter {
+            log_config = log_config.with_filter(FilterBuilder::new().parse(&filter).build())
+        }
+
+        android_logger::init_once(log_config);
+    }
+}
+
+#[cfg(target_os = "ios")]
+mod ios {
+    #[uniffi::export]
+    fn setup_tracing(configuration: String) {
+        tracing_subscriber::registry()
+            .with(EnvFilter::new(configuration))
+            .with(fmt::layer().with_ansi(false).with_writer(io::stderr))
+            .init();
+    }
+}
+
+#[cfg(not(any(target_os = "ios", target_os = "android")))]
+mod other {
+    use std::io;
+
+    use tracing_subscriber::{fmt, prelude::*, EnvFilter};
+
+    #[uniffi::export]
+    fn setup_tracing(configuration: String) {
+        tracing_subscriber::registry()
+            .with(EnvFilter::new(configuration))
+            .with(fmt::layer().with_ansi(false).with_writer(io::stderr))
```

Comment:
If we're going to have an other here we might want to enable colors. AFAIR they are problematic for iOS while they do work on Android and probably for other platforms as well.

Reply:
I just used the existing as a fallback to not have compile errors for the host-platform. Sure we can improve that if you want. just a `with_ansi(true)` then?

Reply:
```suggestion
            .with(fmt::layer().with_ansi(true).with_writer(io::stderr))
```

Reply:
> I just used the existing as a fallback to not have compile errors for the host-platform. Sure we can improve that if you want. just a `with_ansi(true)` then?

Yep.